packages/react-keyring/src/providers/Keyring.tsx

```diff
@@ -81,37 +78,36 @@ export function KeyringProvider ({
   servicePrincipal,
   connection
 }: KeyringProviderProps): JSX.Element {
-  const [agent, setAgent] = useState<Agent>()
+  const [client, setClient] = useState<Client>()
   const [account, setAccount] = useLocalStorageState<string>(W3UI_ACCOUNT_LOCALSTORAGE_KEY)
   const [space, setSpace] = useState<Space>()
   const [spaces, setSpaces] = useState<Space[]>([])
-  const [issuer, setIssuer] = useState<Signer>()
   const [registerAbortController, setRegisterAbortController] =
     useState<AbortController>()
 
-  const getAgent = async (): Promise<Agent> => {
-    if (agent == null) {
-      const a = await createAgent({ servicePrincipal, connection })
-      setAgent(a)
-      setIssuer(a.issuer)
-      setSpace(getCurrentSpaceInAgent(a))
-      setSpaces(getSpaces(a))
-      return a
+  const getClient = async (): Promise<Client> => {
+    if (client == null) {
+      const client = await createClient({ servicePrincipal, connection })
+      setClient(client)
+      setSpace(client.currentSpace())
+      setSpaces(client.spaces())
+      return client
     }
-    return agent
+    return client
   }
 
   const authorize = async (email: `${string}@${string}`): Promise<void> => {
-    const agent = await getAgent()
+    const c = await getClient()
     const controller = new AbortController()
     setRegisterAbortController(controller)
 
     try {
-      await accessAuthorize(agent, email, { signal: controller.signal })
+      await c.authorize(email, { signal: controller.signal })
```

Comment:
Did this not change to `login`?

Reply:
HM you're right - it looks like `authorize` wasn't removed, but I do think this should use login - will update!